### ISSUE TITLE
Unify callbacks with block transfer and non-blocktransfer

### ISSUE THREAD

**Issue by mlasch** [eclipse/wakaama#537]
Currently for each operation there are two possible callbacks, either with or without block transfer. However it is not always known a priori whether a block transfer callback is needed.
For example a "normal" read `lwm2m_dm_read` can induce a block response which triggers callbacks without block information.
I suggest having a block information structure in every callback which keeps track on the block transfer and is NULL if the callback results from a non-block transfer. Something like:
```
typedef struct _block_information_
{
  int block_num;
  int block_size;
  bool block_more;
} block_information_t;
```
However this would break the current API, any thoughts?

**Comment by sbernard31** [comment]
About breaking the API, [this discussion](https://github.com/eclipse/wakaama/issues/487) makes me feel that this is probably not a problem at least for master.

**Comment by tuve** [comment]
I think having a unifed api with an optional struct as described is much better than the current solution. The reason I didn't do something like that when I implemented block transfer was that I didn't wanted to break the current api, but moving forward I think this is the way to go.

**Comment by sbernard31** [comment]
Fix by #540.
